### DISCUSSION TITLE
[V0.1400.0.0] Cargo missions feedback and issues

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#5783] [Unstable feedback]
The new cargo missions in theory use a "Unit Load Device", and spawn several crates in those ULD, with a reward scaling based on the amount of crates. 

Let's start with the good:
Missions finally pay (sometimes).  Being potentially paid 12500mks for a mission justifies having enough storage space and using a transport sub. Transport subs earning a lot of money finally means there's a reason to buy and switch subs? Having opportunity cargo and switching to a transport to make a big buck seems like a viable strategy.

The bad:
**Missions are still uneventful. The most important point, needs to be said twice. See at the bottom**

Balance:
The prices need rebalancing. **Rewards seem highly inconsistent**. I've gotten [1000mks](https://media.discordapp.net/attachments/668819422919524352/842380380115566642/unknown.png), [2400mks](https://cdn.discordapp.com/attachments/668819422919524352/842380678900219924/unknown.png) (20 crates!) and [12500mks ](https://media.discordapp.net/attachments/668819422919524352/842372258776219678/unknown.png) on the Berilia. The only mission that was remotely dangerous (water-reactant) [only paid 2500mks! ](https://cdn.discordapp.com/attachments/668819422919524352/842385429512060978/unknown.png).

Reward scaling per crate doesn't match the sheer amount of cargo space transport subs have. The Berilia has EIGHT ULDs, and the R29 has room for five, but the missions only fill 25 crates at most. There's still a lot of empty storage.
Meanwhile, any medium to large size sub will easily be able to pack two ULDs, almost matching the maximum reward for transport subs. **Cargo missions need to pack significantly more crates** (upper limit on Berilia is 80, maybe that's too much) with less reward per crate so that transport subs still have an advantage

Reward scaling also has to match difficulty. Transport subs are supposedly bad at combat, but it's too easy to run a transport sub in early biomes where enemies are weak and sparse. When you pick cargo missions, it also means you're not picking combat missions, so swarms or molochs will be unlikely to spawn as well. **Early biome cargo shouldn't reward as much as late biome cargo.** The reward needs a biome multiplier as well. 
There's also the more topic of campaign budgeting and balance, it's a thing in SP so you can't start with a Berilia right away, but in MP anything goes.

Other issues also involve the nature of the cargo itself; **we're able to use and deconstruct the cargo**, which leads to earning through using the cargo for your own supply than getting the mission reward. Medical transport? Deconstruct saline, morphine and fentanyl, or even use it to heal yourself if you're doing a Nest mission. Mineral transport? Free stuff!

Being able to pick two or more missions at once means **Cargo as secondary mission is "free money"**. Do any mission you'd do normally, and get your reward for just driving to the exit of the cave. There's no reason to not take Cargo, always. 

You can cheese even further: **maximizing reward by removing crate from shelves** before starting the mission

---

**Missions are still uneventful:**
This means the mission structure of cargo missions is that **cargo missions have no mission structure**. Cargo missions essentially ask you to drive from starting point to finish. This is the exact same as "No Mission", and missions being boring is a capital sin.

There are only three exceptions:
-The crate contains volatile explosives (nitroglycerin, unstable compound N). Both got changed so that explosives can't actually explode (for understable reasons but still) 
-Only one mission so far had Sodium/Potassium, meaning you must protect the cargo from water
-If you have traitors on, they can use items from the cargo

So, apart from the Sodium/Potassium transport, all cargo missions have *nothing* going on, besides piloting the transport sub itself, which may or may not be as bad as the R-29. 

**Cargo missions need to be more interesting that a one-cave drive**.
This can be achieved in three ways:

1. **Tending to the cargo, diversifying the contents**:
Taking a page from Into The Abyss mod, [the cargo isn't passive and reacts to the environment ](https://cdn.discordapp.com/attachments/426913209602146315/778442170377175050/gif_captiveraptor1.gif). Mudraptors can wake up, contents can spoil or break... The boxes matter and there's more than passively carrying from outpost to outpost.  
The cargo itself needs to be a more active task overall, with more diverse types of items. 
And since we're getting pirates this update, consider the eventual boarding shuttle if you're carrying valuable military items...

2. **Mission structure(s) inside the cave**
There needs to be more than just driving from point A to point B. One way is to make the delivery active:
![image](https://user-images.githubusercontent.com/73229309/118141819-13375c80-b40a-11eb-8ebb-87f5d5abfde8.png)
This could be one (of several) cargo mission objectives. Beacon/Station construction sites at the bottom/ceiling of the cave itself. This implies swimming outside to deliver, using shuttles as a way to quickly unload crates on construction sites, which is an active process.

This could be anything really, the point is: _make things happen during Transport missions_ 
See more concepts in https://github.com/Regalis11/Barotrauma/discussions/4971.

3. **Cargo as a non-mission activity** (across outposts / free trade / minus rep for using items)
Instead of using triggering cargo as a mission, allow us to freely buy cargo at outposts, and freely deliver cargo to places that need it. 
Buying bulks of crates at a lower price, and delivering them *across several stations* or points of interest (construction sites!)
Earn money reputation, or trigger special events relating to the type of cargo in the ULD
Transport subs would be better at hoarding large quantities of crates for several trips. This is a "natural economy" approach, instead of simply being paid for the service of immediate delivery. 
In a nutshell, less "Uber Eats" and more "International Shipping"

**Comment by ubertpendragon** [comment]
Moreover, with the ability to take multiple missions at once

There is no reason to not take cargo every time you have the opportunity.
You can also empty Crate Shelves to maximize the amount of loaded cargo before starting the missions. 

Cargo missions are a fast-track to getting money and rep without doing anything

**Comment by ViktorShield** [comment]
I'm just wondering what's the problem about just implementing features like ones described here https://github.com/Regalis11/Barotrauma/discussions/4971 and in ITA mod. There already are all the concepts. Different types of crates and mechanics(time-crates, explosive crates, nearly-hatched eggs, passeger transport, delivery etc.). But yet we're about to have an absolutely boring way to complete campaign. Congratulations on that, really. It's so original just to buff crate capacity for vessels with a specifil label! I am already waiting for your next cool features, like a god damn underwater laser, because who cares
